### PR TITLE
Added animation implementation in react-reveal

### PR DESCRIPTION
**Pull request by karminkarmen**
https://trello.com/c/fieF5PdO/377-imagewidget-allow-animate-option

Added animation attribute using `animate.css` and `react reveal`  library

### CODE REVIEW

File: src/Widgets/ImageWidget/ImageWidgetClass.js [hunk 1]
```diff
@@ -6,6 +6,18 @@ const ImageWidget = Scrivito.provideWidgetClass("ImageWidget", {
     alignment: ["enum", { values: ["left", "center", "right"] }],
     alternativeText: "string",
     link: "link",
+    animation: [
+      "enum",
+      {
+        values: [
```
Comment (apepper) [inline src/Widgets/ImageWidget/ImageWidgetClass.js]: I would also add the value `"none"`.

Please make this also the default value:

```js
  initialContent: {
    animation: "none",
  },
```

File: src/Widgets/ImageWidget/ImageWidgetEditingConfig.js [hunk 2]
```diff
@@ -24,8 +24,21 @@ Scrivito.provideEditingConfig("ImageWidget", {
       title: "Link (optional)",
       description: "The link where this image should lead.",
     },
+    animation: {
+      title: "Animation (optional)",
```
Comment (apepper) [inline src/Widgets/ImageWidget/ImageWidgetEditingConfig.js]: If "None" is an option, we should remove the `(optional)` part.

File: src/Widgets/ImageWidget/ImageWidgetEditingConfig.js [hunk 3]
```diff
@@ -24,8 +24,21 @@ Scrivito.provideEditingConfig("ImageWidget", {
...
+      description:
+        "What animation should be shown, when this image becomes visible?",
+      values: [
+        { value: "", title: "None" },
```
Comment (apepper) [inline src/Widgets/ImageWidget/ImageWidgetEditingConfig.js]: `{ value: "none", title: "None" },`

File: src/Widgets/ImageWidget/ImageWidgetEditingConfig.js [hunk 4]
```diff
@@ -24,8 +24,21 @@ Scrivito.provideEditingConfig("ImageWidget", {
       title: "Link (optional)",
       description: "The link where this image should lead.",
     },
+    animation: {
+      title: "Animation (optional)",
+      description:
+        "What animation should be shown, when this image becomes visible?",
```
Comment (apepper) [inline src/Widgets/ImageWidget/ImageWidgetEditingConfig.js]: `...when this image becomes visible? Default: None`

File: src/Components/AnimateOnReveal.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import * as React from "react";
+import Zoom from "react-reveal/Zoom";
+import Fade from "react-reveal/Fade";
+
+function AnimateOnReveal({ animation, children }) {
```
Comment (apepper) [inline src/Components/AnimateOnReveal.js]: I did think again about the implementation of `AnimateOnReveal`. My next iteration is now this:

```js
import * as React from "react";
import Zoom from "react-reveal/Zoom";
import Fade from "react-reveal/Fade";

function AnimateOnReveal({ animation, children }) {
  switch (animation) {
    case "zoomIn":
      return <Zoom>{children}</Zoom>;
    case "fadeInLeft":
      return <Fade left>{children}</Fade>;
    case "fadeInRight":
      return <Fade right>{children}</Fade>;
    case "fadeInDown":
      return <Fade top>{children}</Fade>;
    case "fadeInUp":
      return <Fade bottom>{children}</Fade>;
    default:
      return children;
  }
}

export default AnimateOnReveal;
```

I think it is a bit clearer to read and also has a better default - rendering the children. E.g. if someone gives a not yet supported animation prop (e.g. "none") the code just works.

**Comment by karminkarmen** [comment]
REVIEW (c2017ae) /cc @apepper

**Comment by karminkarmen** [comment]
REVIEW (f125618) /cc @apepper

**Pull request merged**